test/e2e/opm_test.go

```diff
@@ -435,3 +439,92 @@ var _ = Describe("opm", func() {
 		IncludeSharedSpecs("podman")
 	})
 })
+
+var _ = Describe("can integrate the latest registry image", func() {
+	IncludeSharedSpecs := func(containerTool string) {
+		BeforeEach(func() {
+			if redhatRegistryUsername == "" || redhatRegistryPassword == "" {
+				Skip("redhat registry credentials are not available")
+			}
+
+			dockerlogin := exec.Command(containerTool, "login", "-u", redhatRegistryUsername, "-p", redhatRegistryPassword, "registry.redhat.io")
+			err := dockerlogin.Run()
+			Expect(err).NotTo(HaveOccurred(), "Error logging into registry.redhat.io")
+		})
+
+		It("can integrate with the existing production index image", func() {
+			// This test ensures that opm is compatible with the given production index image
+			// copy the version of opm and serve the prod index contents with it -- check that the index image starts
+			// Ensures that no changes are introduced into operator-registry that cause a problem for indexes
+			By("copying the prod index db out of the index image")
+
+			const (
+				prodIndexName = "registry.redhat.io/redhat/redhat-operator-index:v4.6"
+			)
+
+			dbDirectory, err := ioutil.TempDir("./", "db_temp_")
+			Expect(err).ToNot(HaveOccurred())
+			defer os.RemoveAll(dbDirectory)
+
+			// extract db file from index to the dbDirectory
+			db, err := indexer.ExtractDatabase(prodIndexName, dbDirectory, "", containerTool, true)
+			Expect(err).ToNot(HaveOccurred())
+
+			GinkgoT().Logf("database file %s", db)
+
+			By("using opm registry serve to serve the latest index content")
+			opmRegistryCmd := command.NewOpmRegistryCmd()
+
+			// find registry serve command
+			var serveCmd cobra.Command
+			for _, c := range opmRegistryCmd.Commands() {
+				if c.Use == "serve" {
+					serveCmd = *c
+				}
+			}
+			Expect(serveCmd).ToNot(BeNil())
+
+			// remove default ginkgo args (--test.timeout)
+			serveCmd.SetArgs([]string{})
+			err = serveCmd.Flags().Set("database", db)
+			Expect(err).ToNot(HaveOccurred())
+
+			errChan := make(chan error, 1)
+			go func() {
+				err := serveCmd.RunE(&serveCmd, []string{})
+				if err != nil {
+					errChan <- err
+				}
+			}()
+
+			time.Sleep(1*time.Second)
+			select {
+			case err := <- errChan:
+				Expect(err).ToNot(HaveOccurred())
+			default:
+			}
+
+			// dial server via gRPC client to ensure successful startup
+			// by default server listens on port 50051 for requests
+			var client *rpc.Client
+			Eventually(func() error {
+				const addr = "127.0.0.1:50051"
+				client, err = rpc.Dial("tcp", addr)
+				GinkgoT().Logf("rpc dial error: %s", err)
+				return err
+			}).Should(Succeed())
+
```

Comment:
Can we add some basic sanity checks here?

Maybe:
 - the list bundles api returns something? 
 - the listimages api returns a list of images? (we use it for disconnected mirroring)
 - is there a package we're pretty sure will always exist that we could use? (amq?)

Reply:
I realize that this part of the test will be slightly more brittle, but hopefull if it fails it will always be clear what the issue is, and if the test needs to be updated

Reply:
I think that it's a good idea. I propose using the healthcheck and list bundles endpoints since those together demonstrate the index is running and is the most future-proof (as we move to the declarative-based index format). 

The registry interface exposes these methods so its easy to implement in the test as well.